### PR TITLE
Added Rolling Ops library.

### PR DESCRIPTION
**Pull request by pengale**
This is a first pass at how the Kafka charm might interface with the
Rolling Ops library.

Still TODO:

1) More extensive unit tests (especially testing potentially broken
cases).
2) Discussion in this PR about whether this is what folks were
expecting when they requested a Rolling Ops lib, and whether this
covers the use cases we need it to cover for Kafka.

### CODE REVIEW

File: lib/charms/rolling_ops/v0/rollingops.py [hunk 1]
```diff
@@ -0,0 +1,375 @@
...
+        interface: rolling_op
+```
+
+Import the library, and enable it by doing initializing a RollingOpsManager class, passing
+in the Charm, name of the peer relation, and restart handler.
+
+src/charm.py
+```python
+# ...
+from charms.rolling_ops.v0.rollingops import RollingOpsManager, RollingEvents
```
Comment (pengale) [inline lib/charms/rolling_ops/v0/rollingops.py:38]: Whoops. This import line needs to be updated. (No need to import RollingEvents.)

**Review by marcoppenheimer — APPROVED** [review]
I'm going to approve this, because at first visual pass, I believe this is exactly the implementation needed for this. 

i.e, as I understand it:
```
On `config-changed` event received by a unit, ensure each unit replans one by one before being Ready.
The most likely cause for this event will be an update to the configuration for Kafka's `*.properties`, or a new unit `relation-joined`, both of which are the main scenarios in which we need to rolling restart units within the cluster 
```

If the above is understood, I'm happy to approve this.
I'll hold off merging and testing more thoroughly as I'm not working heavily on this charm right now, but will keep here to revisit.

**Comment by marcoppenheimer** [comment]
This is such a useful library @pengale, thank you!

**Comment by marcoppenheimer** [comment]
@pengale - Looks great, and I'm aware of the change due to Mia's work, thank you!
@davigar15 - You happy for me to merge this, but have it be promoted to `edge` later down the line?

**Comment by pengale** [comment]
I notice that one of the integration tests failed. They are passing (against microk8s) when I run `tox -e integration` locally, though, so I'm guessing that this is a test runner issue, rather than an issue w/ the code.

**Review by delgod — APPROVED** [review]

**Pull request closed without merging**